catroid/src/main/java/org/catrobat/catroid/pocketmusic/note/NoteName.java

```diff
@@ -22,22 +22,27 @@
  */
 package org.catrobat.catroid.pocketmusic.note;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public enum NoteName {
-	A0(21, false), A0S(22, true), B0(23, false), C1(24, false), C1S(25, true), D1(26, false), D1S(27, true), E1(28,
-			false), F1(29, false), F1S(30, true), G1(31, false), G1S(32, true), A1(33, false), A1S(34, true), B1(35,
-			false), C2(36, false), C2S(37, true), D2(38, false), D2S(39, true), E2(40, false), F2(41, false), F2S(42,
-			true), G2(43, false), G2S(44, true), A2(45, false), A2S(46, true), B2(47, false), C3(48, false), C3S(49,
-			true), D3(50, false), D3S(51, true), E3(52, false), F3(53, false), F3S(54, true), G3(55, false), G3S(56,
-			true), A3(57, false), A3S(58, true), B3(59, false), C4(60, false), C4S(61, true), D4(62, false), D4S(63,
-			true), E4(64, false), F4(65, false), F4S(66, true), G4(67, false), G4S(68, true), A4(69, false), A4S(70,
-			true), B4(71, false), C5(72, false), C5S(73, true), D5(74, false), D5S(75, true), E5(76, false), F5(77,
-			false), F5S(78, true), G5(79, false), G5S(80, true), A5(81, false), A5S(82, true), B5(83, false), C6(84,
-			false), C6S(85, true), D6(86, false), D6S(87, true), E6(88, false), F6(89, false), F6S(90, true), G6(91,
-			false), G6S(92, true), A6(93, false), A6S(94, true), B6(95, false), C7(96, false), C7S(97, true), D7(98,
-			false), D7S(99, true), E7(100, false), F7(101, false), F7S(102, true), G7(103, false), G7S(104, true), A7(
-			105, false), A7S(106, true), B7(107, false), C8(108, false);
+	A0(21, false), A0S(22, true), B0(23, false), C1(24, false), C1S(25, true), D1(26, false), D1S(27, true),
```

Comment:
sorry for the late comeback, i thought about this a tad more, and this enum isnt necessary in the first place. midi 0 is a c.
you need only the octave and the note to uniquely identify every single midi code. (like on a chessboard, you wouldnt create an enum with 64 entries with boolean flags determining the color, you only need to know which row, which column)
then you can get everything you need with one liners:

octave = the chosen octave (0-8)
note = the chosen note (0-10)

get midi code:
int midiCode = octave*12 + note;
bool octaveStart = note == 0;
bool octaveStart = midiCode % 12 == 0;

get black piano keys:
static List<Integer> sharpKeys = Arrays.asList(new String[]{1,3,6,8,10});
int sharpKey = sharpKeys.contains(note)
int sharpKey = sharpKeys.contains(midi % 12)

this would be open to extension when eg. adding / modifying avalibale octaves or even adding another format aside midi, bseides requiring half the lines of code.

Reply:
yeah, you are write. I will change that, but we need to keep the existing bounds for the notes, since pianos (the instrument we are using for now) supports only midi from 21 - 108, for reference:
https://newt.phys.unsw.edu.au/jw/notes.html
Additionally to make it easier we only show full octaves, which is why we start with 24 and go up to 108.

As an addtional note I will also remove some of the neighboring enums, which aren't used for anything.